packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -108,72 +146,79 @@ class AndroidValidator extends DoctorValidator {
   Future<ValidationResult> validate() async {
     final List<ValidationMessage> messages = <ValidationMessage>[];
 
-    if (androidSdk == null) {
+    if (_androidSdk == null) {
       // No Android SDK found.
-      if (globals.platform.environment.containsKey(kAndroidHome)) {
-        final String androidHomeDir = globals.platform.environment[kAndroidHome];
-        messages.add(ValidationMessage.error(userMessages.androidBadSdkDir(kAndroidHome, androidHomeDir)));
+      if (_platform.environment.containsKey(kAndroidHome)) {
+        final String androidHomeDir = _platform.environment[kAndroidHome];
+        messages.add(ValidationMessage.error(_userMessages.androidBadSdkDir(kAndroidHome, androidHomeDir)));
       } else {
-        messages.add(ValidationMessage.error(userMessages.androidMissingSdkInstructions(kAndroidHome)));
+        // Instruct user to set [kAndroidSdkRoot] and not deprecated [kAndroidHome]
+        // See https://github.com/flutter/flutter/issues/39301
+        messages.add(ValidationMessage.error(_userMessages.androidMissingSdkInstructions(kAndroidSdkRoot)));
```

Comment:
This is the actual fix. The rest is context refactoring.

Reply:
Should we want people if they've set ANDROID_HOME instead of ANDROID_SDK_ROOT?

Reply:
I don't think it's necessary? According to the [original fix](https://github.com/flutter/flutter/pull/25221), we still check `ANDROID_HOME` because the SDK still does https://developer.android.com/studio/command-line/variables.

Reply:
SGTM